File: diff5/problem25inc/Fibonacci.py
def toFib(index):
    if index == 1:
        return 1
    if index == 2:
        return 1 
    else:
        return toFib(index - 1) + toFib(index - 2)
     
def solution():
    place = 1
    while not digitCount(toFib(place), 1000): 
        place += 1
    print place 
   
def digitCount(number, digits):
    return len(str(number)) == digits 
    
solution()

File: diff5/problem25inc/Fibonacci.java
public class Fibonacci {
    public static void main(String[] commandLine) {
        int place = 1;
        for(; !digitCount(toFib(place),1000);place++){ }
        System.out.println(place);
    }
    public static int toFib(int index) {
        if (index == 1 || index == 2) return 1;
        return toFib(index - 2) + toFib(index - 1);
    }
    public static boolean digitCount(int num, int digits) {
        return num / Math.pow(10, digits - 1) >= 1;
    }
}